fendermint/vm/interpreter/src/fvm/state/query.rs

```diff
@@ -142,15 +146,17 @@ where
     ///
     /// The results are never going to be flushed, so it's semantically read-only,
     /// but it might write into the buffered block store the FVM creates. Running
-    /// multiple such messages results in their buffered effects stacking up.
-    pub fn call(&self, mut msg: FvmMessage) -> anyhow::Result<ApplyRet> {
+    /// multiple such messages results in their buffered effects stacking up if
+    /// `use_cache` is enabled. If `use_cache` is not enabled, the results of the
+    /// execution are not cached.
+    pub fn call(&self, mut msg: FvmMessage, use_cache: bool) -> anyhow::Result<ApplyRet> {
         // If the sequence is zero, treat it as a signal to use whatever is in the state.
         if msg.sequence.is_zero() {
             if let Some((_, state)) = self.actor_state(&msg.from)? {
                 msg.sequence = state.sequence;
             }
         }
```

Comment:
For correctness this should be inside the `with_exec_state`, otherwise the state root can be different once `use_cache: true` has been used to execute some prior transactions. It's my bug and we can fix it in another PR.

Reply:
https://github.com/consensus-shipyard/fendermint/issues/176

Reply:
You are right! Up to you I can fix it here if needed (although it may be cleaner to just start a new PR and get this over with :) )